src/scib_metrics/_pcr.py

```diff
@@ -0,0 +1,80 @@
+from typing import Optional
+
+import jax
+import jax.numpy as jnp
+
+from scib_metrics.utils import one_hot, pca
+
+from ._types import NdArray
+
+
+def principal_component_regression(
+    X: NdArray,
+    covariate: NdArray,
+    categorical: bool = False,
+    n_components: Optional[int] = None,
+) -> float:
+    """Principal component regression (PCR).
+
+    Computes the overall variance contribution given a covariate according to the following formula:
+
+    # TODO: Add formula
+
+    Parameters
+    ----------
+    X
+        Array of shape (n_samples, n_features).
+    covariate
+        Array of shape (n_samples,) or (n_samples, 1) representing batch/covariate values.
+    categorical
+        If True, batch will be treated as categorical and one-hot encoded.
+    n_components:
+        Number of components to compute, passed into :func:`~scib_metrics.utils.pca`. If None, all components are used.
+
+    Returns
+    -------
+    pcr: float
+        Principal component regression using the first n_components principal components.
+    """
+    if len(X.shape) != 2:
+        raise ValueError("Dimension mismatch: X must be 2-dimensional.")
+    if X.shape[0] != covariate.shape[0]:
+        raise ValueError("Dimension mismatch: X and batch must have the same number of samples.")
+
+    covariate = one_hot(covariate) if categorical else covariate.reshape((covariate.shape[0], 1))
+
+    pca_results = pca(X, n_components=n_components)
```

Comment:
realizing that if X is already low-dim we won't see much of a speedup here, so we could opt to use sklearn pca if it helps reproduce their metric

Reply:
Resolving per comment below.